chain/beacon/drand/drand.go

```diff
@@ -239,13 +242,42 @@ var _ beacon.RandomBeacon = (*DrandBeacon)(nil)
 
 func BeaconScheduleFromDrandSchedule(dcs dtypes.DrandSchedule, genesisTime uint64, ps *pubsub.PubSub) (beacon.Schedule, error) {
 	shd := beacon.Schedule{}
-	for _, dc := range dcs {
+	for i, dc := range dcs {
 		bc, err := NewDrandBeacon(genesisTime, buildconstants.BlockDelaySecs, ps, dc.Config)
 		if err != nil {
-			return nil, xerrors.Errorf("creating drand beacon: %w", err)
+			fmt.Printf("failed to create drand beacon: %+v\n", dc)
+			return nil, xerrors.Errorf("%d creating drand beacon: %w", i, err)
 		}
 		shd = append(shd, beacon.BeaconPoint{Start: dc.Start, Beacon: bc})
 	}
 
 	return shd, nil
 }
+
+var _ dclient.Client = HistoricalBeaconClient{}
+
+// HistoricalBeaconClient is a drand client that doesn't actually do anything. It's used when
+// we don't have a drand network to connect to but still need to provide a beacon client.
+// We don't expect calls through to the client to be made since we should only be verifying old
+// randomness, not fetching it.
+type HistoricalBeaconClient struct{}
```

Comment:
```suggestion
type historicalBeaconClient struct{}
```

no need to be public